src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -300,6 +300,19 @@ class PostgresEntityDataQueryService(
         )
     }
 
+    fun getEntityKeyIdsInEntitySet(entitySetId: UUID): Set<UUID> {
```

Comment:
Why convert to a set here? Just use distinct in sql query if you need ids to be distinct.

Reply:
They are distinct anyways, since this is only getting the ids of from entity_key_ids table. It needs to be a set later on in the call stack, when it is called from DataController

Reply:
Use a collection with less overhead (or even a PostgresIterable)

Reply:
That is the way you are using this it could return hundred of millions of UUIDs so it needs to be streaming.